File: Programmers/크레인 인형뽑기 게임.cpp
#include <vector>
#include <stack>
#include <iostream>

using namespace std;

int solution(vector<vector<int>> board, vector<int> moves) {
	int answer = 0;
	stack<int> s;
	for (int i = 0; i < moves.size(); i++)
	{
		int m = moves[i] - 1;
		for (int j = 0; j < board.size(); j++)
		{
			if (board[j][m] != 0) {
				if (!s.empty() && s.top() == board[j][m]){
					s.pop();
                    answer+=2;
                    }
				else s.push(board[j][m]);
				board[j][m] = 0;
                break;
			}
		}
	}
	return answer;
}

File: Programmers/크레인 인형뽑기 게임.java
import java.util.Stack;

class Solution {
	public int solution(int[][] board, int[] moves) {
		int answer = 0;
		Stack<Integer> stack = new Stack<>();
		for (int i = 0; i < moves.length; i++) {

			for (int j = 0; j < board.length; j++) {
				if (board[j][moves[i]-1] == 0)
					continue;
				else {
					if (!stack.empty() && stack.peek() == board[j][moves[i]-1]) {
						stack.pop();
						board[j][moves[i]-1] = 0;
						answer += 2;
						break;
					} else {
						stack.push(board[j][moves[i]-1]);
						board[j][moves[i]-1] = 0;
						break;
					}
				}
			}
		}
		return answer;
	}
}